### PR TITLE
Exosuit Update 2: I Don't Remember Everything I Did Edition

### PR DESCRIPTION
**Pull request by SapphicOverload**
# Document the changes in your pull request

This is somewhat of an extension of my previous PR but this time with a new weapon and significant refactor to how mech destruction and repair are handled in the backend and fixes a number of bugs.

Clarke and Ripley now have a base punch damage of 10 instead of 5, since I thought it was a bit weird that a mech can't punch as hard as a human. Now they can. The Clarke also now has a slightly higher base melee armor (20->25) and can be reinforced up to 50 melee with weaver chitin, similar to how the Ripley can be upgraded from 40 to 70 melee armor with goliath hide plates.

Also makes combat mechs affected by demolition mod again with the modifier being capped at 2x damage from melee weapons, except the supercharged vxtvul hammer.

New Equipment:
- Magnetic Treads - Works like magboots but for mechs.
- Kinetic Crusher - Exosuit equivalent of the handheld kinetic crusher.
- Hydraulic Scythe - Cuts vines in front of the mech.
- Afterburner - Launches the mech forward, mining rocks and damaging anything that gets hit based on punch damage. 10 second cooldown, stronger against lava/jungleland mobs.

Existing Equipment:
- Gravitational Catapult - Now uses right click to lock on and left click to throw, added some visual effects to the repulse mode.
- Repair Droid - Added action button to toggle repair mode.

# Why is this good for the game?

Makes the Clarke, the exosuit specifically designed for mining, less bad at mining by improving its survivability against lavaland mobs.

Also adds more utility to mechs to make them more useful outside of combat.

And the afterburner was just too cool of idea for me not to make it.

# Testing

ROCKET MECH FIGHTING A DRAGON

https://github.com/user-attachments/assets/4cb8593a-e39b-43db-83c9-51cd348077d1

# Spriting

image

image

# Wiki Documentation

Add the new equipment (see above) to the list of equipment on the guide to robotics and update the clarke to 25 base melee armor.

# Changelog

:cl:  
rscadd: Added a few new mech equipment: Magtreads, Kinetic Crusher, Hydraulic Scythe, Afterburners
tweak: Repair droid now uses an action button to toggle
tweak: Gravitational catapult now uses right click to lock on and left click to throw
tweak: Increased base punch damage of Ripley and Clarke from 5 to 10 damage
tweak: Clarke can now attach weaver chitin as armor plating similar to the Ripley with goliath hides
tweak: All mechs are now affected by demolition mod again, caps demolition mod from melee weapons against combat mechs to 2x
bugfix: Fixed mechs with omnidirectional attacks not being able to attack omnidirectionally
bugfix: Fixed hitting a mech with goliath plates or weaver chitin when reinforcing the armor
bugfix: Fixed a ton of other stuff, I don't remember what
imageadd: Adds sprites for new mech equipment and action buttons
spellcheck: Fixed the Durand's defense mode being spelled wrong causing the action button sprite to be invisible
experimental: Refactored mech destruction and repair
experimental: Probably breaks a few things
/:cl:

### CODE REVIEW

**Comment by UselessTheremin** [comment]
great to see the mech code getting some love, and these features look very fun (actually makes me want to play mining with a mech)

File: code/datums/components/afterimage.dm [hunk 1]
```diff
@@ -31,16 +31,15 @@
 
 /datum/component/after_image/RegisterWithParent()
 	loop_timer = addtimer(CALLBACK(src, PROC_REF(spawn_image)), rest_time, TIMER_LOOP|TIMER_UNIQUE|TIMER_STOPPABLE)//start loop
-	RegisterSignal(parent, COMSIG_MOB_CLIENT_PRE_MOVE, PROC_REF(update_step))
-	RegisterSignal(parent, COMSIG_MOB_CLIENT_MOVED, PROC_REF(update_glide))
+	RegisterSignal(parent, COMSIG_MOVABLE_PRE_MOVE, PROC_REF(update_step))
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(update_glide))
```
Comment (Moltijoe) [inline code/datums/components/afterimage.dm:35]: Make sure this doesnt fuck up the sandevistan visual effect
I remember having to do some fucky stuff for it to work

Comment (iamthesilly) [reply code/datums/components/afterimage.dm:35]: It still worked when I tested it earlier.

**Review by Mqiib — APPROVED** [review]
My only issue is the mag treads but i cant say i have any idea what would be fitting anyways and it's already just part of a sprite so whatever I guess

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**